corehq/apps/sms/handlers/form_session.py

```diff
@@ -51,16 +52,23 @@ def form_session_handler(v, text, msg):
             session.modified_time = datetime.utcnow()
             session.save()
 
+            # fetch subevent pk to link inbound sms to
+            try:
+                subevent_id = MessagingSubEvent.objects.get(xforms_session_id=session.pk).pk
```

Comment:
It seems a bit wasteful to pull a whole `MessageSubEvent` row when you only need it's key. This could be changed to a values query to avoid that.
```suggestion
                subevent_id = MessagingSubEvent.objects.values_list("id", flat=True).get(xforms_session_id=session.pk)
```

The second question I have is if there is ever more than one `MessagingSubEvent` for a given `xforms_session_id`? If yes, then that situation will need to be handled too.

Reply:
Offlined with @millerdev, but tests have been added for creating a link to a subevent if it exists, as well as asserting a link does not exist if a subevent does not exist. 

To answer the second question about what to do if multiple subevents are linked to a session, a method on `SQLXFormsSession` already handles this by returning the first entry: https://github.com/dimagi/commcare-hq/blob/4e19d22c5e2a0cb60c6345c9286e9eb165a3965b/corehq/apps/smsforms/models.py#L144-L147

I have updated `form_session.py` to use this method instead.